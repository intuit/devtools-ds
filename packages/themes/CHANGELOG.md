# v1.2.0 (Fri Apr 22 2022)

:tada: This release contains work from a new contributor! :tada:

Thank you, Adrian de la Rosa ([@mormubis](https://github.com/mormubis)), for all your work!

#### 🚀 Enhancement

- table: make the header sticky [#23](https://github.com/intuit/devtools-ds/pull/23) ([@mormubis](https://github.com/mormubis) [@tylerkrupicka](https://github.com/tylerkrupicka))

#### 🐛 Bug Fix

- css magic to fix sticky header borders in FF ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### Authors: 2

- Adrian de la Rosa ([@mormubis](https://github.com/mormubis))
- Tyler Krupicka ([@tylerkrupicka](https://github.com/tylerkrupicka))

---

# v1.0.4 (Sun Apr 25 2021)

:tada: This release contains work from a new contributor! :tada:

Thank you, Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie)), for all your work!

#### 🐛 Bug Fix

- Fix: Make Components Work with SSR [#18](https://github.com/intuit/devtools-ds/pull/18) ([@hipstersmoothie](https://github.com/hipstersmoothie))
- enable canary ([@hipstersmoothie](https://github.com/hipstersmoothie))
- make it ssr friendly ([@hipstersmoothie](https://github.com/hipstersmoothie))

#### Authors: 1

- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))

---

# v1.0.3 (Thu Feb 25 2021)

#### 🐛 Bug Fix

- Fix theme generation bug [#17](https://github.com/intuit/devtools-ds/pull/17) ([@tylerkrupicka](https://github.com/tylerkrupicka))
- fix ua parsing bug stopping theme generation ([@tylerkrupicka](https://github.com/tylerkrupicka))
- Perf: Better Tree Shaking [#16](https://github.com/intuit/devtools-ds/pull/16) ([@tylerkrupicka](https://github.com/tylerkrupicka))
- remove ua parser dep ([@tylerkrupicka](https://github.com/tylerkrupicka))
- stop autothemeprovider from leaking dependencies ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### Authors: 1

- Tyler Krupicka ([@tylerkrupicka](https://github.com/tylerkrupicka))

---

# v1.0.0 (Tue Feb 02 2021)

#### 💥 Breaking Change

- Initial release [#8](https://github.com/intuit/devtools-ds/pull/8) ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### 🐛 Bug Fix

- make chrome default theme ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### Authors: 1

- Tyler Krupicka ([@tylerkrupicka](https://github.com/tylerkrupicka))

---

# v0.130.5 (Sun Jan 24 2021)

#### ⚠️ Pushed to `master`

- Merge branch 'master' of https://github.com/intuit/devtools-ds ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### Authors: 1

- Tyler Krupicka ([@tylerkrupicka](https://github.com/tylerkrupicka))

---

# v0.130.4 (Sun Jan 24 2021)

#### ⚠️ Pushed to `master`

- Merge branch 'master' of https://github.com/intuit/devtools-ds ([@tylerkrupicka](https://github.com/tylerkrupicka))

#### Authors: 1

- Tyler Krupicka ([@tylerkrupicka](https://github.com/tylerkrupicka))

---

# vnull (Sun Jan 24 2021)

#### 🐛 Bug Fix

- Bump version to: v0.130.1 \[skip ci\] (https://github.com/tylerkrupicka)

#### ⚠️ Pushed to `master`

- set access in package json (https://github.com/tylerkrupicka)

#### Authors: 1

- Tyler Krupicka (https://github.com/tylerkrupicka)
