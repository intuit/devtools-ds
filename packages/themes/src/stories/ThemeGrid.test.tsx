import React from "react";
import { render } from "@testing-library/react";

import { ThemeGrid } from "./ThemeGrid";
import { chrome } from "../themes";
import styles from "./ThemeGrid.css";

describe("ThemeGrid", () => {
  test("Create Component with chrome theme", () => {
    const name = "Chrome";
    const { container } = render(<ThemeGrid name={name} theme={chrome} />);

    const themeElements = container.getElementsByClassName(styles.theme);
    const themeHeadings = container.getElementsByTagName("h1");

    expect(themeElements).toHaveLength(1);
    expect(themeHeadings[0].textContent).toBe(`${name} - Light`);
    expect(themeHeadings[1].textContent).toBe(`${name} - Dark`);
  });
  test("It matches the snapshot", () => {
    const { container } = render(<ThemeGrid name="Chrome" theme={chrome} />);
    expect(container).toMatchSnapshot();
  });
});
