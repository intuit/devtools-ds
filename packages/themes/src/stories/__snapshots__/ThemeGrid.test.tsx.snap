// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeGrid It matches the snapshot 1`] = `
<div>
  <div
    class="theme"
  >
    <div>
      <h1>
        Chrome
         - Light
      </h1>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(255, 255, 255);"
          />
          <div
            class="label"
          >
            <h2>
              backgroundColor
            </h2>
            <h4>
              #FFFFFF
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(236, 57, 65);"
          />
          <div
            class="label"
          >
            <h2>
              error
            </h2>
            <h4>
              #EC3941
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(254, 241, 240);"
          />
          <div
            class="label"
          >
            <h2>
              errorBackground
            </h2>
            <h4>
              #FEF1F0
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(254, 215, 214);"
          />
          <div
            class="label"
          >
            <h2>
              errorBorder
            </h2>
            <h4>
              #FED7D6
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(255, 0, 0);"
          />
          <div
            class="label"
          >
            <h2>
              errorText
            </h2>
            <h4>
              #FF0000
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(48, 57, 66);"
          />
          <div
            class="label"
          >
            <h2>
              textColor
            </h2>
            <h4>
              #303942
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(245, 189, 0);"
          />
          <div
            class="label"
          >
            <h2>
              warning
            </h2>
            <h4>
              #F5BD00
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(255, 251, 229);"
          />
          <div
            class="label"
          >
            <h2>
              warningBackground
            </h2>
            <h4>
              #FFFBE5
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(255, 245, 194);"
          />
          <div
            class="label"
          >
            <h2>
              warningBorder
            </h2>
            <h4>
              #FFF5C2
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(92, 61, 0);"
          />
          <div
            class="label"
          >
            <h2>
              warningText
            </h2>
            <h4>
              #5C3D00
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(237, 242, 252);"
          />
          <div
            class="label"
          >
            <h2>
              blue01
            </h2>
            <h4>
              #EDF2FC
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(207, 232, 252);"
          />
          <div
            class="label"
          >
            <h2>
              blue02
            </h2>
            <h4>
              #CFE8FC
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(26, 115, 232);"
          />
          <div
            class="label"
          >
            <h2>
              blue03
            </h2>
            <h4>
              #1A73E8
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(54, 70, 182);"
          />
          <div
            class="label"
          >
            <h2>
              blue04
            </h2>
            <h4>
              #3646B6
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(45, 11, 201);"
          />
          <div
            class="label"
          >
            <h2>
              blue05
            </h2>
            <h4>
              #2D0BC9
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(153, 68, 0);"
          />
          <div
            class="label"
          >
            <h2>
              brown01
            </h2>
            <h4>
              #994400
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(243, 243, 243);"
          />
          <div
            class="label"
          >
            <h2>
              gray01
            </h2>
            <h4>
              #F3F3F3
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(233, 233, 233);"
          />
          <div
            class="label"
          >
            <h2>
              gray02
            </h2>
            <h4>
              #E9E9E9
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(205, 205, 205);"
          />
          <div
            class="label"
          >
            <h2>
              gray03
            </h2>
            <h4>
              #CDCDCD
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(192, 192, 192);"
          />
          <div
            class="label"
          >
            <h2>
              gray04
            </h2>
            <h4>
              #C0C0C0
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(142, 142, 142);"
          />
          <div
            class="label"
          >
            <h2>
              gray05
            </h2>
            <h4>
              #8E8E8E
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(68, 68, 68);"
          />
          <div
            class="label"
          >
            <h2>
              gray06
            </h2>
            <h4>
              #444444
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(49, 118, 50);"
          />
          <div
            class="label"
          >
            <h2>
              green01
            </h2>
            <h4>
              #317632
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(174, 151, 187);"
          />
          <div
            class="label"
          >
            <h2>
              purple01
            </h2>
            <h4>
              #AE97BB
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(189, 132, 194);"
          />
          <div
            class="label"
          >
            <h2>
              purple02
            </h2>
            <h4>
              #BD84C2
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(136, 18, 128);"
          />
          <div
            class="label"
          >
            <h2>
              purple03
            </h2>
            <h4>
              #881280
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(207, 75, 73);"
          />
          <div
            class="label"
          >
            <h2>
              red01
            </h2>
            <h4>
              #CF4B49
            </h4>
          </div>
        </li>
      </ul>
    </div>
    <div>
      <h1>
        Chrome
         - Dark
      </h1>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(37, 37, 37);"
          />
          <div
            class="label"
          >
            <h2>
              backgroundColor
            </h2>
            <h4>
              #252525
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(236, 57, 65);"
          />
          <div
            class="label"
          >
            <h2>
              error
            </h2>
            <h4>
              #EC3941
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(41, 0, 0);"
          />
          <div
            class="label"
          >
            <h2>
              errorBackground
            </h2>
            <h4>
              #290000
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(92, 1, 0);"
          />
          <div
            class="label"
          >
            <h2>
              errorBorder
            </h2>
            <h4>
              #5C0100
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(253, 126, 126);"
          />
          <div
            class="label"
          >
            <h2>
              errorText
            </h2>
            <h4>
              #FD7E7E
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(237, 238, 238);"
          />
          <div
            class="label"
          >
            <h2>
              textColor
            </h2>
            <h4>
              #EDEEEE
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(245, 189, 0);"
          />
          <div
            class="label"
          >
            <h2>
              warning
            </h2>
            <h4>
              #F5BD00
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(51, 43, 0);"
          />
          <div
            class="label"
          >
            <h2>
              warningBackground
            </h2>
            <h4>
              #332B00
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(102, 85, 0);"
          />
          <div
            class="label"
          >
            <h2>
              warningBorder
            </h2>
            <h4>
              #665500
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(253, 252, 248);"
          />
          <div
            class="label"
          >
            <h2>
              warningText
            </h2>
            <h4>
              #FDFCF8
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(154, 187, 219);"
          />
          <div
            class="label"
          >
            <h2>
              blue01
            </h2>
            <h4>
              #9ABBDB
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(89, 166, 201);"
          />
          <div
            class="label"
          >
            <h2>
              blue02
            </h2>
            <h4>
              #59A6C9
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(14, 99, 156);"
          />
          <div
            class="label"
          >
            <h2>
              blue03
            </h2>
            <h4>
              #0E639C
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(9, 61, 105);"
          />
          <div
            class="label"
          >
            <h2>
              blue04
            </h2>
            <h4>
              #093D69
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(25, 37, 56);"
          />
          <div
            class="label"
          >
            <h2>
              blue05
            </h2>
            <h4>
              #192538
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(200, 200, 200);"
          />
          <div
            class="label"
          >
            <h2>
              gray01
            </h2>
            <h4>
              #C8C8C8
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(183, 183, 183);"
          />
          <div
            class="label"
          >
            <h2>
              gray02
            </h2>
            <h4>
              #B7B7B7
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(132, 132, 132);"
          />
          <div
            class="label"
          >
            <h2>
              gray03
            </h2>
            <h4>
              #848484
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(60, 61, 60);"
          />
          <div
            class="label"
          >
            <h2>
              gray04
            </h2>
            <h4>
              #3C3D3C
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(42, 42, 45);"
          />
          <div
            class="label"
          >
            <h2>
              gray05
            </h2>
            <h4>
              #2A2A2D
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(32, 32, 32);"
          />
          <div
            class="label"
          >
            <h2>
              gray06
            </h2>
            <h4>
              #202020
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(242, 151, 102);"
          />
          <div
            class="label"
          >
            <h2>
              orange01
            </h2>
            <h4>
              #F29766
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(238, 137, 83);"
          />
          <div
            class="label"
          >
            <h2>
              orange02
            </h2>
            <h4>
              #EE8953
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(246, 217, 248);"
          />
          <div
            class="label"
          >
            <h2>
              pink01
            </h2>
            <h4>
              #F6D9F8
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(171, 152, 172);"
          />
          <div
            class="label"
          >
            <h2>
              pink02
            </h2>
            <h4>
              #AB98AC
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(208, 214, 251);"
          />
          <div
            class="label"
          >
            <h2>
              purple01
            </h2>
            <h4>
              #D0D6FB
            </h4>
          </div>
        </li>
        <li
          class="item"
        >
          <div
            class="box"
            style="background-color: rgb(152, 127, 253);"
          />
          <div
            class="label"
          >
            <h2>
              purple02
            </h2>
            <h4>
              #987FFD
            </h4>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
